packages/babel-parser/src/parser/expression.js

```diff
@@ -526,7 +532,23 @@ export default class ExpressionParser extends LValParser {
       }
     }
 
-    return this.parseUpdate(node, update, refExpressionErrors);
+    const expr = this.parseUpdate(node, update, refExpressionErrors);
+
+    const startsExpr = this.hasPlugin("v8intrinsic")
+      ? this.state.type.startsExpr
+      : this.state.type.startsExpr && !this.match(tt.modulo);
+    if (
+      isAwait &&
+      startsExpr &&
+      !this.isAmbiguousAwait() &&
+      (this.scope.inFunction ||
+        (this.hasPlugin("topLevelAwait") && !this.inModule))
```

Comment:
I think we don't need extra scope / module checks here. When `this.isAmbiguousAwait()` is `false`. It is no more a valid case of `await` being an identifier so we can just reinterpret as a _potential_ await expression.